be/src/storage/task/engine_checksum_task.cpp

```diff
@@ -99,9 +94,22 @@ Status EngineChecksumTask::_compute_checksum() {
         return st;
     }
 
+    // Calculate the chunk size based on the memory limit of the current thread.
```

Comment:
why calculate the chunk size?
better use `min(config::vector_chunk_size, value_by_limit)` to prevent using more memory than before.